tests/infrastructure/golden_images/update_boot_source/utils.py

```
@@ -234,3 +242,69 @@ def wait_for_created_volume_from_data_import_cron(custom_data_source: DataSource
+    try:
+        for sample in TimeoutSampler(
+            wait_timeout=TIMEOUT_30SEC,
+            sleep=TIMEOUT_5SEC,
+            func=lambda: data_source.source.name != volume_name,
+        ):
+            if sample:
+                raise ResourceValueError(
+                    f"DataSource {data_source.name} volume reference was updated, "
+                    f"expected {volume_name}, "
+                    f"spec: {data_source.instance.spec}"
+                )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)